crates/hyperswitch_domain_models/src/api.rs

```diff
@@ -0,0 +1,105 @@
+use std::fmt::Display;
+
+use common_utils::{
+    events::{ApiEventMetric, ApiEventsType},
+    impl_misc_api_event_type,
+};
+
+#[derive(Debug, Eq, PartialEq)]
+pub enum ApplicationResponse<R> {
+    Json(R),
+    StatusOk,
+    TextPlain(String),
+    JsonForRedirection(api_models::payments::RedirectionResponse),
+    Form(Box<RedirectionFormData>),
+    PaymentLinkForm(Box<PaymentLinkAction>),
+    FileData((Vec<u8>, mime::Mime)),
+    JsonWithHeaders((R, Vec<(String, masking::Maskable<String>)>)),
+    GenericLinkForm(Box<GenericLinks>),
+}
+
+impl<T: ApiEventMetric> ApiEventMetric for ApplicationResponse<T> {
+    fn get_api_event_type(&self) -> Option<ApiEventsType> {
+        match self {
+            Self::Json(r) => r.get_api_event_type(),
+            Self::JsonWithHeaders((r, _)) => r.get_api_event_type(),
+            _ => None,
+        }
+    }
+}
+
+impl_misc_api_event_type!(PaymentLinkFormData, GenericLinkFormData);
+
+#[derive(Debug, Eq, PartialEq)]
+pub struct RedirectionFormData {
+    pub redirect_form: crate::router_response_types::RedirectForm,
+    pub payment_method_data: Option<api_models::payments::PaymentMethodData>,
+    pub amount: String,
+    pub currency: String,
+}
+
+#[derive(Debug, Eq, PartialEq)]
+pub enum PaymentLinkAction {
+    PaymentLinkFormData(PaymentLinkFormData),
+    PaymentLinkStatus(PaymentLinkStatusData),
+}
+
+#[derive(Debug, Eq, PartialEq, Clone, serde::Serialize, serde::Deserialize)]
+pub struct PaymentLinkFormData {
+    pub js_script: String,
+    pub css_script: String,
+    pub sdk_url: String,
+    pub html_meta_tags: String,
+}
+
+#[derive(Debug, Eq, PartialEq, Clone, serde::Serialize, serde::Deserialize)]
+pub struct PaymentLinkStatusData {
+    pub js_script: String,
+    pub css_script: String,
+}
+
+#[derive(Debug, Eq, PartialEq)]
+pub enum GenericLinks {
+    ExpiredLink(GenericExpiredLinkData),
+    PaymentMethodCollect(GenericLinkFormData),
+    PayoutLink(GenericLinkFormData),
+    PayoutLinkStatus(GenericLinkStatusData),
+    PaymentMethodCollectStatus(GenericLinkStatusData),
+}
+
+impl Display for Box<GenericLinks> {
```

Comment:
This could have been implemented on `GenericLinks` itself, `Box<T>` implements `Display` if `T: Display`. Can take up in separate PR.

Reply:
Done